kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -33,17 +33,24 @@ public class KafkaAgent {
     private static final String YAMMER_METRICS_IN_KAFKA_3_2_AND_EARLIER = "kafka.metrics.KafkaYammerMetrics";
 
     private final File sessionConnectedFile;
-    private File brokerReadyFile;
+    private final File brokerReadyFile;
     private MetricName brokerStateName;
     private Gauge brokerState;
     private MetricName sessionStateName;
     private Gauge sessionState;
+    private Boolean inKRaftMode = false;
```

Comment:
Do we need `Boolean` instead of `boolean`? Also, should we set it to `false` in the existing constructor and mark it as final? (no idea if it adds anything, just wondering since you seem to do it to the other fields :-o)

Reply:
I have no idea why I randomly chose Boolean here, that definitely wasn't an intentional choice, and yeah I think using a final variable would be better, I'll update it